src/Libraries/RevitNodes/Elements/DetailCurve.cs

```diff
@@ -0,0 +1,162 @@
+            
+            // Get exsiting element
+            var element = ElementBinder.GetElementFromTrace<Autodesk.Revit.DB.DetailCurve>(document);
+
+            if (element == null)
+            {
+                if (document.IsFamilyDocument)
+                {
+                    document.FamilyCreate.NewDetailCurve(view, curve);
+                }
+                else
+                {
+                    document.Create.NewDetailCurve(view, curve);
+                }                
+            }
+            else
+            {
+                element.SetGeometryCurve(curve, true);                
+            }
+
+            InternalSetCurveElement(element);
+
+            // Commit transaction
+            TransactionManager.Instance.TransactionTaskDone();
+            ElementBinder.SetElementForTrace(this.InternalElement);
+        }
+
+        #endregion
+
+        #region Public static constructors
+
+        /// <summary>
+        /// Construct a Revit DetailCurve element from a curve
+        /// </summary>
+        /// <param name="view">View to place the detail curve on</param>
+        /// <param name="curve">Curve to create detailcurve from</param>
+        /// <returns></returns>
+        public static DetailCurve ByCurve(Revit.Elements.Views.View view, Autodesk.DesignScript.Geometry.Curve curve)
+        {
+            if (!view.IsAnnotationView())
+            {
+                throw new Exception(Properties.Resources.ViewDoesNotSupportAnnotations);
+            }
+
+            Autodesk.Revit.DB.View revitView = (Autodesk.Revit.DB.View)view.InternalElement;
+
+            return new DetailCurve(revitView, curve.ToRevitType(true));
+        }
+
+        #endregion
+
+        #region Public properties
+
+        /// <summary>
+        /// Get Geometry Curve
+        /// </summary>
+        public Autodesk.DesignScript.Geometry.Curve Curve
+        {
+            get
+            { 
+                return this.InternalCurveElement.GeometryCurve.ToProtoType(true);
+            }
+        }
+
+        /// <summary>
+        /// Set Geometry Curve
+        /// </summary>
+        public void SetCurve(Autodesk.DesignScript.Geometry.Curve curve)
+        {
+            Autodesk.Revit.DB.Document document = DocumentManager.Instance.CurrentDBDocument;
+            TransactionManager.Instance.EnsureInTransaction(document);
+            this.InternalCurveElement.SetGeometryCurve(curve.ToRevitType(true),true);
+            TransactionManager.Instance.TransactionTaskDone();
+        }
+
+        #endregion
+
+        #region Private static constructors
+
+        internal static DetailCurve FromExisting(Autodesk.Revit.DB.DetailCurve instance, bool isRevitOwned)
+        {
+            return new DetailCurve(instance)
+            { 
+                IsRevitOwned = isRevitOwned
+            };
+        }
+
+
+        #endregion
+
+        public override string ToString()
```

Comment:
@Randy-Ma Since FromExisting is neither defined in CurveElement nor in Element, I defined it here similar to the other elements